src/analysis/inner_product_space/projection.lean

```diff
@@ -444,6 +444,15 @@ lemma orthogonal_projection_fn_eq (v : E) :
   orthogonal_projection_fn K v = (orthogonal_projection K v : E) :=
 rfl
 
+/-- The orthogonal projection on a submodule `U` of an inner product space `V`,
+as a continuous linear map from `V` to `V`. See also `orthogonal_projection` for the version as
+a continuous linear map from `V` to `U`. -/
+noncomputable def orthogonal_projection.extend (U : submodule 𝕜 E) [complete_space U] :
+  E →L[𝕜] E := U.subtypeL.comp (orthogonal_projection U)
```

Comment:
Is this definition really necessary? My instinct would be that using `U.subtypeL.comp (orthogonal_projection U)` works fine, but if it's too annoying in practice then we can keep it. It's probably better to call it `orthogonal_projection'` though